homeassistant/components/linear_garage_door/config_flow.py

```diff
@@ -0,0 +1,154 @@
+"""Config flow for Linear Garage Door integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+import uuid
+
+from linear_garage_door import Linear
+from linear_garage_door.errors import InvalidDeviceIDError, InvalidLoginError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = {
+    vol.Required("email"): str,
+    vol.Required("password"): str,
+}
+
+
+async def validate_input(
+    hass: HomeAssistant,
+    data: dict[str, Any],
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    hub = Linear()
+
+    device_id = data.get("device_id", str(uuid.uuid4()))
+
+    try:
+        await hub.login(
+            data["email"],
+            data["password"],
+            device_id=device_id,
+            client_session=async_get_clientsession(hass),
+        )
+    except InvalidLoginError as err:
+        raise InvalidAuth from err
+    except InvalidDeviceIDError as err:
+        raise InvalidDeviceID from err
```

Comment:
You already have specific exceptions, you may as well just have them raise and then catch them outside your validate function rather than having an extra intermediary exception in my opinion. This is probably fine as is though.